### PR TITLE
Improve `Runtime` docs

### PR DESCRIPTION
**Pull request by slowli**
This PR clarifies recent changes to `Runtime` interface.

### CODE REVIEW

File: exonum/src/runtime/mod.rs [hunk 1]
```diff
@@ -12,77 +12,85 @@
...
-//! dispatcher to register the artifact as deployed. After that the validators can send requests to
-//! start new service instances from this artifact.
+//! # Service lifecycle
 //!
-//! 3. To start a new service instance, each validator administrator should send a request
-//! to the dispatcher. Each request contains exactly the same artifact identifier, instance name, and
-//! instance configuration parameters. Then, as in the previous case, if the number of
-//! confirmations is equal to the total number of validators, each validator calls the dispatcher
-//! to start a new service instance.
+//! 1. An artifact with the service is deployed on the blockchain. The decision to deploy an
```
Comment (popzxc) [inline exonum/src/runtime/mod.rs]: Maybe it makes sense to add a "0." note explaining where deployable artifacts come from (short & abstract)?

File: exonum/src/runtime/mod.rs [hunk 2]
```diff
@@ -12,77 +12,85 @@
...
+//!   artifact and the deployment spec are usually performed by the blockchain administators.
+//!   The corresponding logic is customizable via a [supervisor service](#supervisor-service).
 //!
-//! 4. Modify instance configuration procedure. TODO [ECR-3306]
+//! 2. Once the artifact is deployed, it is possible to instantiate service.
+//!   Each start request contains the same artifact identifier, instance IDs (the numeric one
```
Comment (popzxc) [inline exonum/src/runtime/mod.rs]: Does request contain numeric ID? I thought it's assigned by the blockchain.

File: exonum/src/runtime/mod.rs [hunk 3]
```diff
@@ -12,77 +12,85 @@
...
-//! 1. An Exonum client creates a transaction message which includes [CallInfo] information
-//! about the corresponding method to call and serialized method parameters as a payload.
-//! The client then signs the message with the author's key pair.
+//! 1. An Exonum client creates a transaction message which includes [`CallInfo`] information
+//!   about the corresponding method to call and serialized method parameters as a payload.
+//!   The client then signs the message with the author's key pair.
 //!
 //! 2. The client transmits the message to one of the Exonum nodes in the network.
-//! The transaction is identified by the hash of the corresponding message.
+//!   The transaction is identified by the hash of the corresponding message.
```
Comment (popzxc) [inline exonum/src/runtime/mod.rs]: Maybe it worth noting that hash is calculated from the signed message.

File: exonum/src/runtime/mod.rs [hunk 4]
```diff
@@ -12,77 +12,85 @@
...
+//!   immediate [execution].
 //!
 //! 6. After execution the transaction [execution status] is written into the blockchain.
 //!
-//! # Service interfaces
+//! # Supervisor Service
 //!
-//! In addition to its own methods, a service can implement methods of additional interfaces.
-//! In your own runtime implementation, you must provide support of the following interfaces:
+//! A supervisor service is a service that has additional privileges: it can command deployment of
```
Comment (popzxc) [inline exonum/src/runtime/mod.rs]: I think that configuration changes should also be mentioned here.

File: exonum/src/runtime/mod.rs [hunk 5]
```diff
@@ -12,77 +12,85 @@
...
+//! artifacts and instantiation of services. Other than that, it is an ordinary service.
+//! A supervisor should be present on blockchain start, otherwise no new artifacts / services
+//! could ever be added to the blockchain.
 //!
-//! TODO: Think about runtime agnostic interfaces description. [ECR-3531]
+//! A supervisor service is distinguished by its numerical ID, which must be set
```
Comment (popzxc) [inline exonum/src/runtime/mod.rs]: Maybe add a note why it should not be referenced by its name?

**Comment by aleksuss** [comment]
@slowli take a look to description for `on_resume` method. `after_initialize` method does not exist anymore.

File: exonum/src/runtime/mod.rs [hunk 6]
```diff
@@ -12,77 +12,85 @@
...
-//! start new service instances from this artifact.
+//! # Service lifecycle
 //!
-//! 3. To start a new service instance, each validator administrator should send a request
-//! to the dispatcher. Each request contains exactly the same artifact identifier, instance name, and
-//! instance configuration parameters. Then, as in the previous case, if the number of
-//! confirmations is equal to the total number of validators, each validator calls the dispatcher
-//! to start a new service instance.
+//! 1. An artifact with the service is deployed on the blockchain. The decision to deploy an
+//!   artifact and the deployment spec are usually performed by the blockchain administators.
```
Comment (aleksuss) [inline exonum/src/runtime/mod.rs]:
```suggestion
//!   artifact and the deployment spec are usually performed by the blockchain administrators.
```

**Pull request merged**